.github/workflows/python-app.yml

```diff
@@ -1,60 +1,61 @@
+          python-version: 3.8
+          node-version: ${{ env.NODE_VERSION }}
+      - name: Install dependencies Conduct Static analysis Test Coverage Mutation
+        run: |
+          python3 build_scripts/dependencies_static_analysis_test_cov.py
+      - uses: actions/upload-artifact@v2
+        with:
+          name: my-artifact_static_analysis
+          path: |
+            htmlcov
+            CC.xml
+            jscpd-report.json
+            mutation.txt
 
-    runs-on: ubuntu-latest
-
+  publish:
+    name: Build , Smoke test , Performance test, and publish Python 🐍 distributions 📦 to PyPI and TestPyPI
+    needs: build
+    runs-on: ubuntu-18.04
     steps:
     - uses: actions/checkout@v2
-    - name: Set up Python 3.7
+    - name: Fetch all history for all tags and branches
+      run: git fetch --prune --unshallow
+    - name: Set up Python 3.8
       uses: actions/setup-python@v2
       with:
-        python-version: 3.7
-        node-version: ${{ env.NODE_VERSION }}
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
-    - name: npm install
-      run: |
-        sudo npm install
-        sudo npm install -g jscpd@3.2.1
-    - name: ctags install
-      run: |
-        sudo apt-get install ctags
-        export PATH="$PATH:/usr/bin/ctags"
-        ctags --version
-    - name: Static Analysis
-      run: |
-        pylint functiondefextractor/ test/
-        jscpd --min-tokens 20 --reporters "json" --mode "strict" --format "python" -o . .
-        python3 test_resource/jscpd_parser.py --j 10 #Allowed duplication is 10% with 20 token size
-
-    - name: Cyclomatic complexity
+        python-version: 3.8
+    - name: Install the dependencies
       run: |
-        python3 -m lizard functiondefextractor -X> CC.xml
-        python3 test_resource/cyclo_gate.py --c 7 #Allowed cyclomatic complexity is less than 7
-    - name: Dead Code
+        python3 build_scripts/install_dependencies.py
+    - name: Build a binary wheel and a source tarball
       run: |
-        python3 -m vulture functiondefextractor
-    - name: Tests & Coverage
+        python3 setup.py sdist bdist_wheel
+    - name: Install newly created wheel
       run: |
-        pytest test --cov-config=.coveragerc --cov=functiondefextractor
-        coverage report --fail-under=90 #Coverage should be above 90%
-        codecov
-    - name: Mutation Tests
+          python3 build_scripts/install_package.py
+    - name: Conduct the sanity test on new wheel created
       run: |
-        mutmut run > mutmut.log || true # Checking mutation on main file as the travis max execution time is overshooting
-        mutmut junitxml --suspicious-policy=ignore --untested-policy=ignore > mutmut.xml
-        python3 test_resource/mutmut_parse.py --m 20 #Mutation (failure+disables+error) allowed 20% of total mutants
+          python3 build_scripts/sanity_test.py
```

Comment:
newline is missing here. Check if the yml linting is correctly finding it

Reply:
Corrected. Thank You for the comment